st2common/tests/unit/test_jinja_render_regex_filters.py

```diff
@@ -68,3 +68,24 @@ def test_filters_regex_search(self):
         actual = env.from_string(template).render({'k1': 'v0.10.1'})
         expected = 'True'
         self.assertEqual(actual, expected)
+
+    def test_filters_regex_substring(self):
+        env = jinja_utils.get_jinja_environment()
+
+        # Normal (match)
+        template = '{{k1 | regex_substring("(127.0.0.1)")}}'
```

Comment:
Edit: I also see user can pass in `group_index` - would also be good to add a test case when an on-zero group index is used and when there is a match and when there is none.